app/logic/transcript.py

```diff
@@ -10,72 +10,82 @@
 from peewee import DoesNotExist, fn
 
 
-#FIXME: Needs to break hours down by program and term, not just program
+def getOtherEventsTranscript(username):
+    """
+    Returns a Other Events query object containing all the non-program
+    events for the current user.
+    """
+
+    otherEventsData = list(EventParticipant.select(EventParticipant, Event, Program, fn.SUM(EventParticipant.hoursEarned).alias("hoursEarned"))
+                    .join(Event)
+                    .join(ProgramEvent)
+                    .join(Program)
+                    .where(EventParticipant.user == username,
+                           Event.isTraining == False,
+                           Event.isAllVolunteerTraining == False,
+                           ((ProgramEvent.program == None) |
+                            (Program.isStudentLed == False) &
+                            (Program.isBonnerScholars == False)))
+                    .group_by(Event.term)
+                  )
+
+    otherEvents = []
+
+    for term in otherEventsData:
+        otherEvents.append([term.event.term.description, term.hoursEarned])
```

Comment:
A good case for a list comprehension

Reply:
Also, you are currently doing a separate query to get the term description for every entry in the list

Reply:
Fixed the extra query but a list comprehension would likely not be ideal unless we use something like `[x.append('a') or x for x in a]`

Reply:
I was thinking `return [[term.event.term.description, term.hoursEarned] for term in otherEventsData]`